bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/common/WatcherCommon.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.folderwatcher.internal.common;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link WatcherCommon} class contains commonly used methods.
+ *
+ * @author Alexandr Salamatov - Initial contribution
+ */
+@NonNullByDefault
+public class WatcherCommon {
+
+    private static void initFile(File file, String watchDir) throws IOException {
+        try (BufferedWriter fileWriter = new BufferedWriter(new FileWriter(file))) {
+            fileWriter.write(watchDir);
+            fileWriter.newLine();
+        }
+    }
+
+    public static List<String> initStorage(File file, String watchDir) throws IOException {
+        List<String> returnList = new ArrayList<>();
+        List<String> currentFileListing = new ArrayList<>();
```

Comment:
If these variables are just going to get reassigned then they don't need to be ArrayList instances right?
```suggestion
        List<String> returnList = List.of();
        List<String> currentFileListing = List.of();
```

Reply:
fixed